gaia-header.js

```diff
@@ -34,105 +51,270 @@ module.exports = Component.register('gaia-header', {
-    for (var i = 0; i < this.els.headings.length; i++) {
-      fontFit.reformatHeading(this.els.headings[i]);
-    }
+  fontFit: function(text, space) {
+    debug('font fit', text, space);
+    return fontFit({
+      font: 'italic 300 24px FiraSans',
+      space: space,
+      text: text,
+      min: 16,
+      max: 24
+    });
   },
 
   /**
-   * Triggers the 'action' button
-   * (used in testing).
+   * Get the title text for a given
+   * `title` object.
    *
-   * @public
+   * Takes care of caching the last title
+   * text and uses this value to indicate
+   * if it has changed.
+   *
+   * @param  {Object} title
+   * @return {Object}
+   * @private
    */
-  triggerAction: function() {
-    if (this.isSupportedAction(this.getAttribute('action'))) {
-      this.els.actionButton.click();
-    }
+  getTitleText: function(el) {
+    debug('get title text');
+    var previous = el._textContent;
+    var text = el.textContent;
+
+    // Cache it
+    el._textContent = text;
+
+    return {
+      value: text,
+      changed: text !== previous
+    };
+  },
+
+  /**
+   * Start the observer listening
+   * for DOM mutations.
+   *
+   * @private
+   */
+  observerStart: function() {
+    if (this.observing) { return; }
+
+    this.observer.observe(this, {
+      childList: true,
+      attributes: true,
+      subtree: true
+    });
+
+    this.observing = true;
+    debug('observer started');
+  },
+
+  /**
+   * Stop the observer listening
+   * for DOM mutations.
+   *
+   * @private
+   */
+  observerStop: function() {
+    if (!this.observing) { return; }
+    this.observer.disconnect();
+    this.observing = false;
+    debug('observer stopped');
   },
 
   /**
-   * Configure the action button based
-   * on the value of the `data-action`
-   * attribute.
+   * When the components DOM changes we run
+   * the debounced version runFontFit.
    *
+   * @param  {Array} mutations
    * @private
    */
-  configureActionButton: function() {
-    var old = this.els.actionButton.getAttribute('icon');
-    var type = this.getAttribute('action');
-    var supported = this.isSupportedAction(type);
-    this.els.actionButton.classList.remove('icon-' + old);
-    this.els.actionButton.setAttribute('icon', type);
-    this.els.inner.classList.toggle('supported-action', supported);
-    if (supported) { this.els.actionButton.classList.add('icon-' + type); }
+  onMutation: function(mutations) {
+    debug('on mutation', mutations);
+    this.runFontFit();
   },
 
   /**
-   * Validate action against supported list.
+   * Get the title width.
    *
+   * Returns the space available for <h1>.
+   *
+   * @return {[type]} [description]
    * @private
    */
-  isSupportedAction: function(action) {
-    return action && actionTypes[action];
+  getTitleSpace: function() {
+    var titlePadding = 20;
+    var previous = this.titleSpace;
+    var value = this.getWidth()
+      - this.titleStart
+      - this.titleEnd
+      - titlePadding
+      - 1;
```

Comment:
add a comment for this magic value (or even use a local variable with a nice name to hold this value)